plugins/modules/pacemaker_resource.py

```diff
@@ -160,14 +159,13 @@ class PacemakerResource(StateModuleHelper):
             )),
             wait=dict(type='int', default=300),
         ),
-        required_if=[('state', 'present', ['resource_type', 'resource_option'])],
+        required_if=[('state', 'present', ['resource_type', 'resource_option']), ('state', ['present', 'absent', 'enabled', 'disabled'], ['name'])],
```

Comment:
```suggestion
        required_if=[
            ('state', 'present', ['resource_type', 'resource_option']),
            ('state', ['present', 'absent', 'enabled', 'disabled'], ['name']),
        ],
```
Also I'm pretty sure that this won't work. `state` cannot be a list.
```suggestion
        required_if=[
            ('state', 'present', ['resource_type', 'resource_option', 'name']),
            ('state', 'absent', ['name']),
            ('state', 'enabled', ['name']),
            ('state', 'disabled', ['name']),
            ('state', ['present', 'absent', 'enabled', 'disabled'], ['name']),
        ],
```